environment-api/src/main/java/com/sequenceiq/environment/api/v1/environment/model/request/azure/AzureResourceEncryptionParameters.java

```diff
@@ -0,0 +1,82 @@
+package com.sequenceiq.environment.api.v1.environment.model.request.azure;
+
+import javax.validation.constraints.AssertTrue;
+import javax.validation.constraints.Pattern;
+
+import com.sequenceiq.environment.api.doc.environment.EnvironmentModelDescription;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+@ApiModel(value = "AzureResourceEncryptionV1Parameters")
+public class AzureResourceEncryptionParameters {
+
+    @ApiModelProperty(EnvironmentModelDescription.KEY_VAULT_ID)
+    @Pattern(regexp = "(^[a-zA-Z-][0-9a-zA-Z-]*)",
+            message = "Vault name must only contain alphanumeric characters and dashes and cannot start with a number." +
+                    "keyVaultId should be provided along with keyUrl.")
+    private String keyVaultId;
+
+    @ApiModelProperty(EnvironmentModelDescription.KEY_URL)
+    @Pattern(regexp = "(^https:\\/\\/[a-zA-Z-][0-9a-zA-Z-]*\\.vault\\.azure\\.net\\/keys\\/[0-9a-zA-Z-]*)",
```

Comment:
1. Why the need for the matching group encompassing the whole pattern?
2. Why not include the `/<keyVersion>` part as well in the pattern?